tests/lib/compiler-test-derive/src/ignores.rs

```diff
@@ -0,0 +1,195 @@
+use std::fs::File;
+use std::path::PathBuf;
+
+use std::io::{BufRead, BufReader};
+
+pub const CFG_TARGET_OS: &'static str = env!("CFG_TARGET_OS");
+pub const CFG_TARGET_ARCH: &'static str = env!("CFG_TARGET_OS");
+
+#[derive(Debug, Clone)]
+struct IgnorePattern {
+    os: Option<String>,
+    arch: Option<String>,
+    engine: Option<String>,
+    compiler: Option<String>,
+    pattern_to_ignore: String,
+}
+
+impl IgnorePattern {
+    fn should_ignore(
+        &self,
+        os: &str,
+        arch: &str,
+        engine: &str,
+        compiler: &str,
+        canonical_path: &str,
+    ) -> bool {
+        self.os.as_ref().map_or(true, |val| val == os)
+            && self.arch.as_ref().map_or(true, |val| val == arch)
+            && self.engine.as_ref().map_or(true, |val| val == engine)
+            && self.compiler.as_ref().map_or(true, |val| val == compiler)
+            && (self.pattern_to_ignore == "*" || canonical_path.contains(&*self.pattern_to_ignore))
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct Ignores {
+    /// The canonical path, and the set of features
+    patterns: Vec<IgnorePattern>,
+}
+
+impl Ignores {
+    /// If the path matches any of the paths on the list
+    pub fn should_ignore(
+        &self,
+        os: &str,
+        arch: &str,
+        engine: &str,
+        compiler: &str,
+        canonical_path: &str,
+    ) -> bool {
+        self.patterns.iter().any(|p| {
+            // println!(" -> {:?}", p);
+            p.should_ignore(os, arch, engine, compiler, canonical_path)
+        })
+    }
+
+    pub fn should_ignore_host(&self, engine: &str, compiler: &str, canonical_path: &str) -> bool {
+        self.should_ignore(
+            CFG_TARGET_OS,
+            CFG_TARGET_ARCH,
+            engine,
+            compiler,
+            canonical_path,
+        )
+    }
+
+    /// Build a Ignore structure from a file path
+    pub fn build_from_path(path: PathBuf) -> Ignores {
+        let file = File::open(path).unwrap();
+        let reader = BufReader::new(file);
+        let mut patterns = Vec::new();
+
+        for line in reader.lines() {
+            let line = line.unwrap();
+            // If the line has a `#` we discard all the content that comes after
+            let line = if line.contains('#') {
+                let l: Vec<&str> = line.split('#').collect();
+                l.get(0).unwrap().to_string()
```

Comment:
⛳ `l.get(0).unwrap()` -> `l[0]`

Reply:
This pattern appears a number of times but I only called it out once. It's slightly simpler but semantically the same. `get(0).unwrap()` is more explicit to the reader about the potential to failure though (by being more text and grabbing more attention), but otherwise they're the same

Reply:
Good call

Reply:
Fixed